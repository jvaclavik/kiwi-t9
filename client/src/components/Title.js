// @flow
import React from "react"

// components
import { Text, View } from "react-native"

// utils
import renderStyle from "../utils/renderStyle"

// theme
import { Metrics, Fonts, Colors } from "../themes"

const styles = {
  title: {
    fontSize: Fonts.sizes.huge,
    fontWeight: "900",
    color: Colors.text,
  },
  titleContainer: {
    marginHorizontal: Metrics.spacings.section,
    marginTop: Metrics.spacings.large,
    marginBottom: Metrics.spacings.base,
  },
}

type Props = {|
  +children: string,
  style?: StyleValue,
|}

export default class Title extends React.PureComponent<Props> {
  static defaultProps = {
    style: {},
  }

  render() {
    const { children, style } = this.props
    return (
      <View style={renderStyle([styles.titleContainer, style])}>
        <Text style={renderStyle(styles.title)}>{children}</Text>
      </View>
    )
  }
}
