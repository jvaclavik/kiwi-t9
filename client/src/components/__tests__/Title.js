// @flow
import React from "react"
import renderer from "react-test-renderer"

// components
import Title from "../Title"

test("renders correctly with required props only", () => {
  const tree = renderer.create(<Title>Title</Title>).toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders correctly with optional props", () => {
  const tree = renderer
    .create(<Title style={{ backgroundColor: "white" }}>Title</Title>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
