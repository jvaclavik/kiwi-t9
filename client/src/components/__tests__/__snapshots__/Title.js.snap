// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with optional props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "marginHorizontal": 40,
      "marginVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 32,
        "fontWeight": "900",
      }
    }
  >
    Title
  </Text>
</View>
`;

exports[`renders correctly with required props only 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 40,
      "marginVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 32,
        "fontWeight": "900",
      }
    }
  >
    Title
  </Text>
</View>
`;
